sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,354 @@
+     * {@code buildAsyncClient()} is called a new instance of {@link TextAnalyticsAsyncClient} is created.
+     *
+     * <p>
+     * If {@link #pipeline(HttpPipeline) pipeline} is set, then the {@code pipeline} and
+     * {@link #endpoint(String) endpoint} are used to create the {@link TextAnalyticsClient client}. All other builder
+     * settings are ignored.
+     * </p>
+     *
+     * @return A TextAnalyticsAsyncClient with the options set from the builder.
+     * @throws NullPointerException if {@link #endpoint(String) endpoint} or
+     * {@link #subscriptionKey(String) subscriptionKey} has not been set.
+     * @throws IllegalArgumentException if {@link #endpoint(String) endpoint} cannot be parsed into a valid URL.
+     */
+    public TextAnalyticsAsyncClient buildAsyncClient() {
+        // Global Env configuration store
+        final Configuration buildConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration().clone() : configuration;
+        // Service Version
+        final TextAnalyticsServiceVersion serviceVersion =
+            version != null ? version : TextAnalyticsServiceVersion.getLatest();
+
+        // Endpoint cannot be null, which is required in request authentication
+        Objects.requireNonNull(endpoint, "'Endpoint' is required and can not be null.");
+
+        // Create a default Pipeline if it is not given
+        if (httpPipeline == null) {
+            // Closest to API goes first, closest to wire goes last.
+            final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+            // Authentications
+            if (tokenCredential != null) {
+                // User token based policy
+                policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+            } else if (subscriptionKey != null) {
+                headers.put(OCP_APIM_SUBSCRIPTION_KEY, subscriptionKey);
+            } else {
+                // Throw exception that credential and tokenCredential cannot be null
+                logger.logExceptionAsError(
+                    new IllegalArgumentException("Missing credential information while building a client."));
+            }
+
+            policies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
+                buildConfiguration));
+            policies.add(new RequestIdPolicy());
+            policies.add(new AddHeadersPolicy(headers));
+            policies.add(new AddDatePolicy());
+
+            HttpPolicyProviders.addBeforeRetryPolicies(policies);
+            policies.add(retryPolicy == null ? DEFAULT_RETRY_POLICY : retryPolicy);
+            policies.addAll(this.policies);
+            HttpPolicyProviders.addAfterRetryPolicies(policies);
+
+            policies.add(new HttpLoggingPolicy(httpLogOptions));
+
+            httpPipeline = new HttpPipelineBuilder()
```

Comment:
This is setting the class level `HttpPipeline`, shouldn't this use a local variable instead? Any policy changes to this builder won't take affect since it will just use the pipeline constructed here in the future.